packages/token/src/erc20/interface.cairo

```diff
@@ -43,6 +43,20 @@ pub trait IERC20CamelOnly<TState> {
     ) -> bool;
 }
 
+#[starknet::interface]
+pub trait IERC20Permit<TState> {
+    fn permit(
+        ref self: TState,
+        owner: ContractAddress,
+        spender: ContractAddress,
+        amount: u256,
+        deadline: u64,
+        signature: Array<felt252>
```

Comment:
```suggestion
        signature: Span<felt252>
```

Reply:
We have to require a signature in the format of an array

Reply:
See https://github.com/OpenZeppelin/cairo-contracts/pull/1140#discussion_r1790077515